README.md

```diff
@@ -8,14 +8,157 @@ Based on steverhoades/oauth2-openid-connect-client.
+you can get login url by yourself and store state string into session storage `oidc-auth.state`, see that middleware.
 
 ### `config/auth.php`
 
 Set guard driver to `oidc-auth-session` to remove tokens from session storage on logout.
 
-Set provider driver to `oidc-auth-session` to make `Auth::user()` return a `LaravelOIDCAuth\OIDCUser` authenticable from saved token.
+Set provider driver to `none`, user provider is not suitable.
 
-Alternatively, you can implement a user factory (see `LaravelOIDCAuth\UserFactoryInterface`) for things like creating user DB model and use provider driver like `eloquent`.
+Alternatively, you can implement a user factory (see `LaravelOIDCAuth\Contracts\OIDCAuthenticatableFactory`) for things
+like creating user DB model and use provider driver like `eloquent`.
+See [User Model & User Factory](#user-model--user-factory)
+
+## Documentation
+
+### Tokens
+
+You can use authenticated user object to get tokens.
+
+```php
+// Get login user, returns \LaravelOIDCAuth\Contracts\OIDCAuthenticatable|null
+Auth::user();
+// Get the saved access token, returns \OpenIDConnectClient\AccessToken|null
+Auth::user()->accessToken();
+// Get the saved id token, returns \Lcobucci\JWT\Token|null
+Auth::user()->accessToken()->getIdToken();
+// Get the saved refresh token, returns string|null
+Auth::user()->accessToken()->getRefreshToken();
+```
+
+### ID Token
+
+You can get id token claims like this:
+
+```php
+Route::get('/user', function () {
+    return Auth::user()->accessToken()->getIdToken()->getClaims();
+});
+```
+
+### Refresh the access token
+
+If you are using `oidc-auth-session` guard with `OIDCUser`, you can easily refresh the access token:
+
+```php
+use LaravelOIDCAuth\Facades\OIDC;
+// ...
+Auth::user()->refreshAccessToken();
+```
+
+Typically, you don't need to refresh the access token manually, the following doc shows an automatic way to do this.
+
+#### Auto refresh
+
+Sometimes, you want the `oidc-auth-session` to refresh the access token automatically every request, please edit
+`config/oidc_auth.php`:
+
+```php
+    // ...
+    'auto_refresh' => true,
+    // ...
+```
+
+Next time the user logged in, the `CallbackController` will set `rememberMe` option from
+`config('oidc_auth.auto_refresh')` to the guard's `login` function.
+
+### User Model & User Factory
+
+If you need an authenticated user to be an eloquent model with OIDC features, your user model must implement
+`LaravelOIDCAuth\Contracts\OIDCAuthenticatable`:
+
+```php
+namespace App\Models;
+
+use Illuminate\Notifications\Notifiable;
+use Illuminate\Contracts\Auth\MustVerifyEmail;
+use LaravelOIDCAuth\Contracts\OIDCAuthenticatable;
+use LaravelOIDCAuth\HasAccessToken;
+use LaravelOIDCAuth\RefreshesAccessToken;
+use Illuminate\Foundation\Auth\User as Authenticatable;use OpenIDConnectClient\AccessToken;
+
+class User extends Authenticatable implements OIDCAuthenticatable
+{
+    use Notifiable, HasAccessToken, RefreshesAccessToken;
```

Comment:
Why is Notifiable required?

Also, I don't think code example is required here. Mentioning traits that may help implementation is enough.

Reply:
Okay.